x-pack/plugins/data_quality/public/application.tsx

```diff
@@ -44,7 +44,6 @@ const AppWithExecutionContext = ({
 }) => {
   const { executionContext } = core;
 
-  // TODO: Remove this comment after testing
```

Comment:
@maryam-saeidi this override is on purpose as described in 
[[Dataset quality] Adding execution context to data quality page within management app](https://github.com/elastic/kibana/pull/189272).

I removed the comment  with [fix(dataset-quality): remove TODO](https://github.com/elastic/kibana/pull/201118/commits/81a712c71eab969a2a7c7c232315b806e8b849a5)

However, it seems your changes are affecting how the pageName is sent for this app.

Before this PR, the pageName for dataset quality was constantly tracked as `application:management:data_quality`.

<img width="567" alt="Screenshot 2024-11-26 at 12 56 21" src="https://github.com/user-attachments/assets/025b358b-e541-4b37-a002-8179dd18b916">

Visiting the same page on this PR, we get relative paths for the sub-router of this app instead of the target app.

<img width="556" alt="Screenshot 2024-11-26 at 12 57 11" src="https://github.com/user-attachments/assets/c57c0b38-0d58-4af1-b15e-6ab279766dd4">

Can't say exactly what is the turning point, but the values are different now 😕

Reply:
@maryam-saeidi, I think the problem here is that manual useExecutionContext() is called before the ones that are called in Routes, so Route overrides the custom one. Ideally the final one could look like this: "application: management: data_quality:details"